contracts/statechannels/AllocationExchange.sol

```diff
@@ -75,29 +77,26 @@ contract AllocationExchange is Governed {
         emit AuthoritySet(authority);
     }
 
-    function redeem(bytes calldata _msg) public {
-        // Parse voucher
-        AllocationVoucher memory voucher = abi.decode(_msg, (AllocationVoucher));
+    function redeem(AllocationVoucher calldata _voucher) public {
+        require(_voucher.amount > 0, "Exchange: zero tokens voucher");
 
-        // Already collected check
+        // Already redeemed check
         require(
-            !allocationsRedeemed[voucher.allocationId],
-            "Collect: allocation already collected"
+            !allocationsRedeemed[_voucher.allocationID],
+            "Exchange: allocation already redeemed"
         );
 
         // Signature check
-        bytes32 collectHash = keccak256(abi.encodePacked(voucher.amount, voucher.allocationId));
-        require(
-            authority == ECDSA.recover(collectHash, voucher.signature),
-            "Collect: invalid signer"
-        );
+        bytes32 messageHash = keccak256(abi.encodePacked(_voucher.allocationID, _voucher.amount));
+        bytes32 digest = ECDSA.toEthSignedMessageHash(messageHash);
```

Comment:
We don't need this `toEthSignedMessageHash`. As with some of our other designs, this will use a dedicated key that serves no other purpose and holds no funds so there are no collisions. The performance is better this way, and it requires fewer libraries from the signing side. I'm not a big fan of just using `toEthSignedMessageHash` in general, because by itself it only fixes a limited set of collisions. It's an important subset of collisions, but can be a false sense of security without something like EIP-712.

Reply:
+1

Reply:
@That3Percent I removed the prefix from the signature, just a plain one now